src/restapi.rs

```diff
@@ -38,6 +156,44 @@ pub async fn list_advisory_labels(user: &mut GooseUser) -> TransactionResult {
     Ok(())
 }
 
+/// Upload file and return advisory ID
+pub async fn upload_advisory_and_get_id(
+    advisory_file: &str,
+    user: &mut GooseUser,
+) -> Result<String, Box<TransactionError>> {
+    // Check cache first - only cache the base template
+    let base_content = get_base_upload_content(advisory_file)
+        .await
+        .map_err(|e| Box::new(TransactionError::Custom(e)))?;
+
+    // Generate new content with modified ID
+    let generated_content = generate_advisory_content(&base_content)?;
+
+    // Upload the generated content and extract advisory ID
+    upload_advisory_and_extract_id(generated_content, user).await
+}
+
+/// Delete advisory by ID
+pub async fn delete_advisory_by_id(advisory_id: String, user: &mut GooseUser) -> TransactionResult {
+    let uri = format!("/api/v2/advisory/{}", advisory_id);
+    user.delete(&uri).await?;
+    Ok(())
+}
+
+/// Sequential execution: upload and then immediately delete
+pub async fn upload_and_immediately_delete(
+    advisory_file: String,
```

Comment:
Why not load the file on startup, and then pass it in? That would allow dropping all the complex static map and mutex.